src/api/Nhs.Appointments.Core/UserService.cs

```diff
@@ -19,10 +23,20 @@ public Task<string> GetApiUserSigningKey(string clientId)
     {
         return store.GetApiUserSigningKey(clientId);
     }
-    
-    public Task SaveUserAsync(string userId, string scope, IEnumerable<RoleAssignment> roleAssignments)
+
+    public async Task UpdateUserRoleAssignmentsAsync(string userId, string scope, IEnumerable<RoleAssignment> roleAssignments)
+    {
+        var oldRoles = await store.UpdateUserRoleAssignments(userId, scope, roleAssignments);
+
+        var rolesRemoved = oldRoles.Where(old => !roleAssignments.Any(r => r.Role == old.Role));
+        var rolesAdded = roleAssignments.Where(newRole => !oldRoles.Any(r => r.Role == newRole.Role));
+
+        await bus.Send(new UserRolesChanged { User = userId, Added = rolesAdded.Select(r => r.Role).ToArray(), Removed = rolesRemoved.Select(r => r.Role).ToArray()});
```

Comment:
This will send the role ids in the event, but at some point we will want resolve these to the display names for those roles before sending to gov notify so that the user doesn't receive a message containing role ids